native/kotlin/api/kotlin/src/integrationTest/kotlin/CategoriesEndpointTest.kt

```diff
@@ -0,0 +1,131 @@
+package rs.wordpress.api.kotlin
+
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Test
+import uniffi.wp_api.PostListParams
+import uniffi.wp_api.SparseCategoryFieldWithEditContext
+import uniffi.wp_api.CategoryCreateParams
+import uniffi.wp_api.CategoryListParams
+import uniffi.wp_api.CategoryUpdateParams
+import uniffi.wp_api.WpErrorCode
+import uniffi.wp_api.wpAuthenticationFromUsernameAndPassword
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+private const val CATEGORY_ID_48: Long = 48
+private const val CATEGORY_ID_59: Long = 59
+
+class CategoriesEndpointTest {
+    private val testCredentials = TestCredentials.INSTANCE
+    private val siteUrl = testCredentials.parsedSiteUrl
+    private val authentication = wpAuthenticationFromUsernameAndPassword(
+        username = testCredentials.adminUsername, password = testCredentials.adminPassword
+    )
+    private val client = WpApiClient(siteUrl, authentication)
+
+    @Test
+    fun testCategoryListRequest() = runTest {
+        val categoryList = client.request { requestBuilder ->
+            requestBuilder.categories().listWithEditContext(params = CategoryListParams())
+        }.assertSuccessAndRetrieveData().data
+        assert(categoryList.isNotEmpty())
+    }
+
+    @Test
+    fun testFilterCategoryListRequest() = runTest {
+        val categoryList = client.request { requestBuilder ->
+            requestBuilder.categories().filterListWithEditContext(
+                params = CategoryListParams(),
+                fields = listOf(
+                    SparseCategoryFieldWithEditContext.NAME,
+                    SparseCategoryFieldWithEditContext.SLUG
+                )
+            )
+        }.assertSuccessAndRetrieveData().data
+        assert(categoryList.isNotEmpty())
+        assertNull(categoryList.first().description)
+    }
+
+    @Test
+    fun testRetrieveMediaRequest() = runTest {
+        val category = client.request { requestBuilder ->
+            requestBuilder.categories().retrieveWithEditContext(CATEGORY_ID_59)
+        }.assertSuccessAndRetrieveData().data
+        assertNotNull(category)
+    }
```

Comment:
Not sure how I sneaked this one in here. I am removing it in the `/themes` PR I am working on right now.

Reply:
Correction: This is a valid test, the only issue is the name of the function which is corrected in the `/themes` PR.